# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/check-def.sh
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/check-def.sh
--- a/gfx/cairo/cairo/src/check-def.sh	Mon May 06 19:22:35 2024 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,48 +0,0 @@
-#!/bin/sh
-
-LC_ALL=C
-export LC_ALL
-
-if which nm 2>/dev/null >/dev/null; then
-	:
-else
-	echo "'nm' not found; skipping test"
-	exit 0
-fi
-
-test -z "$srcdir" && srcdir=.
-test -z "$MAKE" && MAKE=make
-stat=0
-
-$MAKE check-has-hidden-symbols.i > /dev/null || exit 1
-if tail -1 check-has-hidden-symbols.i | grep CAIRO_HAS_HIDDEN_SYMBOLS >/dev/null; then
-	echo "Compiler doesn't support symbol visibility; skipping test"
-	exit 0
-fi
-
-if [ "`uname -s`" = "Linux" ]; then
-	get_cairo_syms='( objdump -t "$so" | grep "^[^ ]* [^l.*]*[.]"; objdump -t "$so" | grep "[.]hidden.*\\<cairo"; ) | sed "s/.* //"'
-else
-	get_cairo_syms='nm "$so" | grep " [BCDGINRSTVW] " | cut -d" " -f3'
-fi
-
-defs="cairo.def"
-$MAKE $defs > /dev/null
-for def in $defs; do
-	lib=`echo "$def" | sed 's/[.]def$//'`
-	lib=`echo "$lib" | sed 's@.*/@@'`
-	so=.libs/lib${lib}.so
-
-	test -f "$so" || continue
-
-	echo Checking that $so has the same symbol list as $def
-
-	{
-		echo EXPORTS
-		eval $get_cairo_syms | c++filt --no-params | grep -v '^_cairo_test_\|^_fini\|^_init\|^_save[fg]pr\|^_rest[fg]pr\|^_Z\|^__gnu\|^__bss\|^_edata\|^_end' | sort -u
-		# cheat: copy the last line from the def file!
-		tail -n1 "$def"
-	} | diff "$def" - >&2 || stat=1
-done
-
-exit $stat